src/electron/add_tap_device.bat

```diff
@@ -27,12 +27,29 @@ set DEVICE_HWID=tap0901
 set PATH=%PATH%;%SystemRoot%\system32;%SystemRoot%\system32\wbem;%SystemRoot%\system32\WindowsPowerShell/v1.0
 
 :: Check whether the device already exists.
-netsh interface show interface name=%DEVICE_NAME% >nul
+netsh interface show interface name=%DEVICE_NAME%
 if %errorlevel% equ 0 (
   echo TAP network device already exists.
   goto :configure
 )
 
+echo Storing existing network interfaces...
+set NETWORK_INTERFACES_FILE=%tmp%\outlineinstaller-network-interfaces.txt
+:: This command retrieves the existing network interface names and formats it as
+:: a commma-separated string, so it can be passed to find_tap_name.exe:
+::  - removes empty lines with findstr
+::  - removes leading/trailing space with trim
+::  - joins the names without newlines (CLRF) with get-content and set-content
+::  - stores the result in NETWORK_INTERFACES_FILE
+::
+:: Note that we pipe input from /dev/null to prevent Powershell hanging forever
+:: waiting on EOF.
+powershell "wmic nic where 'netconnectionid is not null' get netconnectionid | findstr /r /v '^$' | foreach-object {$_.trim()} > '%NETWORK_INTERFACES_FILE%'; ((get-content '%NETWORK_INTERFACES_FILE%') -join ',') | set-content -nonewline '%NETWORK_INTERFACES_FILE%'" <nul
```

Comment:
Variable expansion is a nightmare in batch. I tried to do the transformation in Go, but ran into encoding issues (for some reason `strings.Trim` fails to remove CRLF). The reason I used a file is because `powershell` cannot export a variable to batch. Additionally, `get-content` (which operates on a file) is the most straightforward way I found to join the names without adding CRLFs.

Reply:
OK, sounds complicated.  I was thinking that even if you have to do the CRLF stripping in powershell, that it might be possible without using a temporary file, but I don't know powershell at all, and that would only be slightly simpler anyway.

Reply:
I wonder why that happened in Go -- I got it to work just fine in the playground: https://play.golang.org/p/hmTgmEamXSk